### PR TITLE
SALTO-4578 - add getInstancesFromElementSource

### PR DESCRIPTION
**Pull request by idoser96**
Added getInstancesFromElementSource to get instance by type from elementSource

---

Also converted all usages
except for contexts_project_filter, as it is a bit more complicated there (and I don't want to overdo it)

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/adapter-utils/src/utils.ts [hunk 1]
```diff
@@ -1323,3 +1323,11 @@ export const formatConfigSuggestionsReasons = (reasons: string[]): string => {
 export const isResolvedReferenceExpression = (value: unknown): value is ReferenceExpression => (
   isReferenceExpression(value) && !(value.value instanceof UnresolvedReference) && value.value !== undefined
 )
+
+export const getInstancesFromElementSource = async (elementSource: ReadOnlyElementsSource, typeNames: string[])
+  : Promise<InstanceElement[]> =>
+  awu(await elementSource.list())
```
Comment (alonstern) [inline packages/adapter-utils/src/utils.ts:1329]: @ori-moisis WDYT? do you have an idea if this is better than doing `getAll` and then filter?

**Review by alonstern — COMMENTED** [review]
thanks 👌

File: packages/adapter-utils/src/utils.ts (hunk 1 above)
Comment (ori-moisis) [reply packages/adapter-utils/src/utils.ts:1329]: did you check the performance?
IIRC last time I checked filtering `getAll` works noticeably faster than many separate `get` calls

File: packages/adapter-utils/src/utils.ts [hunk 2]
```diff
@@ -1323,3 +1323,10 @@ export const formatConfigSuggestionsReasons = (reasons: string[]): string => {
 export const isResolvedReferenceExpression = (value: unknown): value is ReferenceExpression => (
   isReferenceExpression(value) && !(value.value instanceof UnresolvedReference) && value.value !== undefined
 )
+
+export const getInstancesFromElementSource = async (elementSource: ReadOnlyElementsSource, typeNames: string[])
```
Comment (netama) [inline packages/adapter-utils/src/utils.ts:1327]: please don't forget to add a unittest if this is shared logic

**Pull request merged**